akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java

```diff
@@ -368,29 +380,36 @@ public void actorRefWithAckExample() throws Exception {
     Source<String, NotUsed> words =
       Source.from(Arrays.asList("hello", "hi"));
 
+    final TestKit probe = new TestKit(system);
+
     ActorRef receiver =
-        system.actorOf(Props.create(AckingReceiver.class));
+        system.actorOf(Props.create(AckingReceiver.class, probe.getRef()));
 
     Sink<String, NotUsed> sink = Sink.<String>actorRefWithAck(receiver,
         new StreamInitialized(),
-        new Ack(),
+        Ack.INSTANCE,
         new StreamCompleted(),
         ex -> new StreamFailure(ex)
     );
 
     words
         .map(el -> el.toLowerCase())
         .runWith(sink, mat);
+
+    probe.expectMsg("Stream initialized");
+    probe.expectMsg("hello");
+    probe.expectMsg("hi");
+    probe.expectMsg("Stream completed");
     //#actorRefWithAck
   }
 
 
   @Test
   public void callingExternalServiceWithMapAsync() throws Exception {
     new TestKit(system) {
-      final TestProbe probe = new TestProbe(system);
+      final TestKit probe = new TestKit(system);
```

Comment:
Wait, should it really even be changed to a new test kit instance??

Reply:
AFAIR that's the way we wanted to move to in Typed, not sure about changing it here in untyped -- your call @raboof

Reply:
Looking at https://doc.akka.io/docs/akka/current/testing.html#using-multiple-probe-actors?language=java it looked like we were already recommending `TestKit` for Java and `TestProbe` for Scala - that also seemed more native since `TestKit` has a `getRef()` and `TestProbe` only has a `ref()`.

I don't feel strongly either way but this seemed consistent. A quick search shows we seem to have `new TestKit` slightly more often than `new TestProbe`, but it's certainly not consistent ;).

Reply:
Hmm. Messy. @ktoso I think what you are referring to is a different thing, here we are already inside one TestKit instance, and then need a probe, and therefore create a separate TestKit. I think in typed this is much better separated into two different things.

Reply:
Then I'm OK with the change